From 366165fc19e862d08ff51ee2bbbff253fc234dd1 Mon Sep 17 00:00:00 2001
From: Nick Mathewson <nickm@torproject.org>
Date: Fri, 8 Feb 2013 22:10:05 -0500
Subject: [PATCH 2/2] Avoid using $(top_srcdir) in TESTS.

Newer automakes don't like this.

Signed-off-by: Nick Mathewson <nickm@torproject.org>
----
Patch backported from upstream:
https://github.com/libevent/libevent/commit/2863c837007c9f28ae542663a6098c91e704b3ca.patch

Slight modifications to test/Makefile.am

Signed-off-by: Maxime Hadjinlian <maxime.hadjinlian@gmail.com>
---
 Makefile.am      |  4 +++-
 test/Makefile.am | 10 +++++++++-
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index dc90359..c34576d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -128,6 +128,8 @@ endif
 
 SUBDIRS = . include sample
 
+DISTCLEANFILES=
+
 if BUILD_WIN32
 
 SYS_LIBS = -lws2_32 -lshell32 -ladvapi32
@@ -239,5 +241,5 @@ doxygen: FORCE
 	doxygen $(srcdir)/Doxyfile
 FORCE:
 
-DISTCLEANFILES = *~ libevent.pc ./include/event2/event-config.h
+DISTCLEANFILES += *~ libevent.pc ./include/event2/event-config.h
 
diff --git a/test/Makefile.am b/test/Makefile.am
index b10c41a..e2f4f4e 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -5,6 +5,7 @@
 # See LICENSE for copying information.
 
 AUTOMAKE_OPTIONS = foreign
+DISTCLEANFILES=
 
 AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/compat -I$(top_srcdir)/include -I../include -DTINYTEST_LOCAL
 
@@ -19,7 +20,14 @@ endif
 EXTRA_PROGRAMS = regress
 noinst_HEADERS = tinytest.h tinytest_macros.h regress.h tinytest_local.h
 
-TESTS = $(top_srcdir)/test/test.sh
+# We need to copy this file, since automake doesn't want us to use top_srcdir
+# in TESTS.
+TESTS = test/test-script.sh
+
+test/test-script.sh: test/test.sh
+ cp $< $@
+
+DISTCLEANFILES += test/test-script.s
 
 BUILT_SOURCES =
 if BUILD_REGRESS
-- 
2.1.1

